fetch/connection-pool/resources/network-partition-key.py

```diff
@@ -27,17 +25,17 @@ def main(request, response):
     # Unless nocheck_partition is true, check partition_id against server_state, and update server_state.
     stash = request.server.stash
     test_failed = False
-    if request.GET.first(b"nocheck_partition", None) != "True":
+    if request.GET.first(b"nocheck_partition", None) != b"True":
         # Need to grab the lock to access the Stash, since requests are made in parallel.
         with stash.lock:
             # Don't use server hostname here, since H2 allows multiple hosts to reuse a connection.
             # Server IP is not currently available, unfortunately.
-            address_key = str(request.client_address) + "|" + str(request.url_parts.port)
-            server_state = stash.take(uuid) or { "test_failed": False }
-            if address_key in server_state and server_state[address_key] != partition_id:
-                server_state["test_failed"] = True
-            server_state[address_key] = partition_id
-            test_failed = server_state["test_failed"]
+            address_key = str(request.client_address) + u"|" + str(request.url_parts.port)
+            server_state = stash.take(uuid) or {u"test_failed": False}
+            if address_key in server_state and server_state[address_key] != isomorphic_decode(partition_id):
+                server_state[u"test_failed"] = True
+            server_state[address_key] = isomorphic_decode(partition_id)
```

Comment:
Nit: can we directly store and compare the binary `partition_id`?

Reply:
Yes. Code updated.